src/main/java/com/atlassian/db/replica/spi/jdbc/JdbcUrl.java

```diff
@@ -0,0 +1,56 @@
+package com.atlassian.db.replica.spi.jdbc;
+
+public class JdbcUrl {
```

Comment:
Did you think about the changes with Occam's razor in mind? I mean going from String to URI doesn't increase the number of entities, but going further and introducing three entitles just to pass the URI seems to be overkill (especially that it could be handled without the new API/SPI. ConnectionPoolPerReplicaProvider implementation can provide validation specific to the implemented database).